pkgs/build-support/build-typst-package.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  pkgs,
+  stdenvNoCC,
+}:
+{
+  buildInputs ? [ ],
+  typstDeps ? [ ],
+  ...
+}@attrs:
+let
+  cleanAttrs = lib.flip lib.removeAttrs [ "typstDeps" ];
+in
+stdenvNoCC.mkDerivation (
+  (cleanAttrs attrs)
+  // (with attrs; {
+    name = "typst-package-${pname}-${version}";
+
+    buildInputs = buildInputs ++ [ pkgs.yq-go ];
+
+    installPhase =
+      let
+        outDir = "$out/${pname}/${version}";
+      in
+      ''
+        runHook preInstall
+        mkdir -p ${outDir}
+
+        for excluded in $(yq '.package.exclude[]' typst.toml)
```

Comment:
What if that key doesn't exist in the file?

Reply:
Then it does nothing --- treated as an empty array. However, I am expecting `typst.toml` to always exist, as this is part of the requirement of Typst packages